src/world/area_kmr/kmr_06/8D2A10.c

```diff
@@ -14,4 +14,21 @@ INCLUDE_ASM(s32, "world/area_kmr/kmr_06/8D2A10", func_80241A68_8D2F78);
 
 INCLUDE_ASM(s32, "world/area_kmr/kmr_06/8D2A10", func_80241C78_8D3188);
 
-INCLUDE_ASM(s32, "world/area_kmr/kmr_06/8D2A10", func_80241CC8_8D31D8);
+ApiStatus func_80241CC8_8D31D8(ScriptInstance* script, s32 isInitialCall) {
+    // todo use correct types
+    s32* mapVar10 = get_variable(script, SI_MAP_VAR(10));
+    s32 mapVar11 = get_variable(script, SI_MAP_VAR(11));
+    s32 mapVar12 = get_variable(script, SI_MAP_VAR(12));
+
+    func_801235C0(mapVar10[1]);
+
+    heap_free(mapVar10);
+    heap_free(mapVar11);
+    heap_free(mapVar12);
+
+    set_variable(script, SI_MAP_VAR(10), 0);
```

Comment:
I kinda feel like these should be NULL (considering they were pointers), but idk, not sure how I feel.

Reply:
yeah idk because it's unclear what they are and if they're all the same, etc. I feel it should just be revisited and cleaned up later at which point that can be fixed

Reply:
I agree with @pixel-stuck. Once we work out exactly what these are we can add `#define FOO SI_MAP_VAR(10)` etc in the header also